src/unix/tcp.c

```diff
@@ -162,7 +162,7 @@ int uv__tcp_connect(uv_connect_t* req,
     r = connect(uv__stream_fd(handle), addr, addrlen);
   while (r == -1 && errno == EINTR);
 
-  if (r == -1) {
+  if (r == -1 && errno) {
```

Comment:
You should `errno = 0;` before the call to connect() inside the do/while block.  Stylistically, can you check `errno != 0` here?